### PR TITLE
Aggregate aggregated metrics submitted by platforms; aggregate metrics across all platforms

### PR DESCRIPTION
**Pull request by kathy-t**
Corresponding webservice PR: https://github.com/dockstore/dockstore/pull/5486

**Description for [SEAB-5319](https://ucsc-cgl.atlassian.net/browse/SEAB-5319)**
This PR implements the aggregation of aggregated metrics and individual execution metrics because platforms can now submit aggregated metrics.

**Description for [SEAB-5305](https://ucsc-cgl.atlassian.net/browse/SEAB-5305)**
This PR aggregates metrics across all platforms and sends it to Dockstore database using the `PUT aggregatedMetrics` endpoint and `Partner.ALL` value for the `platform` parameter. This PR calculates it by aggregating the aggregated platform metrics. If others feel strongly that it should get all the executions for each platform and aggregate them from scratch, then let me know.

Sample output from running the metrics aggregator locally:
```
Aggregated metrics for tool ID #workflow/github.com/gatk-workflows/seq-format-conversion/BAM-to-Unmapped-BAM, version 3.0.0, platform AGC from directory workflow/github.com/gatk-workflows/seq-format-conversion%2FBAM-to-Unmapped-BAM/3.0.0/
Aggregated metrics for tool ID #workflow/github.com/gatk-workflows/seq-format-conversion/BAM-to-Unmapped-BAM, version 3.0.0, platform TERRA from directory workflow/github.com/gatk-workflows/seq-format-conversion%2FBAM-to-Unmapped-BAM/3.0.0/
Aggregated metrics across all platforms (AGC, TERRA) for tool ID #workflow/github.com/gatk-workflows/seq-format-conversion/BAM-to-Unmapped-BAM, version 3.0.0 from directory workflow/github.com/gatk-workflows/seq-format-conversion%2FBAM-to-Unmapped-BAM/3.0.0/
```

**Review Instructions**
View instructions in https://github.com/dockstore/dockstore/pull/5486

**Issue**
[SEAB-5319](https://ucsc-cgl.atlassian.net/browse/SEAB-5319)
[SEAB-5305](https://ucsc-cgl.atlassian.net/browse/SEAB-5305)

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install` in the project that you have modified (until https://ucsc-cgl.atlassian.net/browse/SEAB-5300 adds multi-module support properly)
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If you are changing dependencies, check with dependabot to ensure you are not introducing new high/critical vulnerabilities
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead. 

[SEAB-5319]: https://ucsc-cgl.atlassian.net/browse/SEAB-5319?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SEAB-5305]: https://ucsc-cgl.atlassian.net/browse/SEAB-5305?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by denis-yuen** [comment]
> A bit of a digression, but how/where will we run this? I think you said somewhere else that the idea is to run it on the deployer, right? If so, how will we get the JAR to the deployer? Build it on the deployer? Pull from Maven (but I don't think it's auto-deployed there)?

Probably release to artifactory and then wget from there from the deployer. 
We don't have the github actions deploy in this repo, but we could (and in the meantime, can still do a normal `mvn release:prepare/perform`)

File: metricsaggregator/pom.xml [hunk 1]
```diff
@@ -37,7 +37,7 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <github.url>scm:git:git@github.com:dockstore/dockstore-support.git</github.url>
         <system-stubs.version>2.0.1</system-stubs.version>
-        <dockstore.core.version>1.14.0-alpha.10</dockstore.core.version>
```
Comment (denis-yuen) [inline metricsaggregator/pom.xml:40]: Hmmmm, a snapshot doesn't make much sense in this context. I should be making a beta/rc release late today

**Review by denis-yuen — COMMENTED** [review]
To be continued

File: metricsaggregator/pom.xml (hunk 1 above)
Comment (kathy-t) [reply metricsaggregator/pom.xml:40]: Oh yes, I will be updating that once the webservice is merged (forgot to leave a TODO)

**Comment by kathy-t** [comment]
> > A bit of a digression, but how/where will we run this? I think you said somewhere else that the idea is to run it on the deployer, right? If so, how will we get the JAR to the deployer? Build it on the deployer? Pull from Maven (but I don't think it's auto-deployed there)?
> 
> Probably release to artifactory and then wget from there from the deployer. We don't have the github actions deploy in this repo, but we could (and in the meantime, can still do a normal `mvn release:prepare/perform`)

In the review instructions for the aggregator PRs, I've been telling people to git clone the repo onto the deployer and building it there, but a release to artifactory sounds more ideal. Is that something that someone with sufficient permissions has to do manually? 

We'll also want this [script ](https://github.com/dockstore/dockstore-support/blob/ce5b1d65a879e36179e50abd81497a9847b6b94f/metricsaggregator/scripts/format-dnastack-validation-data.sh) on the deployer so that we can format the miniwdl data file that DNAstack gave us

**Comment by denis-yuen** [comment]
> Is that something that someone with sufficient permissions has to do manually?

Yeah, I think a few of us have artifactory accounts. Otherwise, could copy https://github.com/dockstore/dockstore/blob/develop/.github/workflows/deploy_tagged.yml#L12-L57 into this repo (and do some hookup)

**Comment by denis-yuen** [comment]
I don't disagree, we'll see where we get to re: more frequent releases

**Comment by kathy-t** [comment]
> > Is that something that someone with sufficient permissions has to do manually?
> 
> Yeah, I think a few of us have artifactory accounts. Otherwise, could copy https://github.com/dockstore/dockstore/blob/develop/.github/workflows/deploy_tagged.yml#L12-L57 into this repo (and do some hookup)

Should I create a separate 1.14 ticket for this? Or is the manual mvn release okay?

**Comment by denis-yuen** [comment]
I'm thinking 1.14.x for getting a github actions release working (i.e. could be part of the release, but doesn't need to block it)

**Review by svonworl — APPROVED** [review]

**Comment by kathy-t** [comment]
> I'm thinking 1.14.x for getting a github actions release working (i.e. could be part of the release, but doesn't need to block it)

I created [SEAB-5423](https://ucsc-cgl.atlassian.net/browse/SEAB-5423) for doing the manual mvn release, which will be done for 1.14. I also created [SEAB-5424](https://ucsc-cgl.atlassian.net/browse/SEAB-5424) to hook up a GitHub Action for a long-term solution

**Review by denis-yuen — APPROVED** [review]

**Pull request merged**